src/App.tsx

```diff
@@ -168,37 +164,37 @@ function App() {
   );
 }
 
-function getMessage(logging: JSON): undefined | getMessageReturn {
-  const data = JSON.parse(JSON.stringify(logging));
+function getMessage(logging: JSON): LogRecord | undefined {
```

Comment:
Related to https://github.com/DiamondLightSource/logpanel/pull/27#discussion_r1601243585

I think you should be able to come up with a class that maps directly to the API response, which hopefully makes this function much shorter.